auth/realm/base/src/test/java/org/wildfly/security/auth/realm/AggregateRealmTest.java

```diff
@@ -0,0 +1,287 @@
+        authenticationAttributes.add("office", 0, "A");
+
+        Attributes authorizationAttributes = new MapAttributes();
+        authorizationAttributes.add("team", 0, "Two");
+
+        Function<Principal, Principal> principalTransformer = new CaseRewriter().asPrincipalRewriter();
+
+        SecurityRealm testRealm = createSecurityRealm(true, authenticationAttributes, principalTransformer, authorizationAttributes);
+        RealmIdentity identity = testRealm.getRealmIdentity(IDENTITY_PRINCIPAL);
+
+        assertTrue("Identity exists", identity.exists());
+
+        Attributes identityAttributes = identity.getAuthorizationIdentity().getAttributes();
+        assertEquals("Expected attribute count.", 1, identityAttributes.size());
+        assertEquals("Expected team", "Two", identityAttributes.get("team", 0));
+    }
+
+    private static SecurityRealm createSecurityRealm(boolean applyPrincipalTransformer, Attributes authentication, Function<Principal, Principal> principalTransformer, Attributes... authorization) {
+        SecurityRealm authenticationRealm = toSecurityRealm(authentication);
+        SecurityRealm[] authorizationRealms = new SecurityRealm[authorization.length];
+        for (int i = 0; i < authorizationRealms.length; i++) {
+            if (authentication == authorization[i]) {
+                authorizationRealms[i] = authenticationRealm;
+            } else {
+                authorizationRealms[i] = applyPrincipalTransformer ? toSecurityRealmWithTransform(authorization[i]) : toSecurityRealm(authorization[i]);
+            }
+        }
+
+        return new AggregateSecurityRealm(authenticationRealm, principalTransformer, authorizationRealms);
+    }
+
+    private static SecurityRealm toSecurityRealm(Attributes attributes) {
+        SimpleMapBackedSecurityRealm securityRealm = new SimpleMapBackedSecurityRealm();
+        if (attributes != null) {
+            Map<String, SimpleRealmEntry> identityMap = new HashMap<>();
+            identityMap.put(IDENTITY_NAME, new SimpleRealmEntry(Collections.emptyList(), attributes));
+            securityRealm.setIdentityMap(identityMap);
+        }
+
+        return securityRealm;
+    }
+
+    private static SecurityRealm toSecurityRealmWithTransform(Attributes attributes) {
+        SimpleMapBackedSecurityRealm securityRealm = new SimpleMapBackedSecurityRealm();
+        if (attributes != null) {
+            Map<String, SimpleRealmEntry> identityMap = new HashMap<>();
+            identityMap.put(IDENTITY_NAME_TRANSFORMED, new SimpleRealmEntry(Collections.emptyList(), attributes));
+            securityRealm.setIdentityMap(identityMap);
+        }
+
+        return securityRealm;
+    }
+
+    /*
+     * Function to convert string to all caps
+     */
+    static class CaseRewriter implements NameRewriter {
+        public String rewriteName(String original) {
+            if (original == null ) return original;
+            return original.toUpperCase(Locale.ROOT);
```

Comment:
```suggestion
            return (original == null) ? null : original.toUpperCase(Locale.ROOT);
```

Reply:
nitpick I know, I just noticed because CaseRewriter class is there twice and ternary might look nicer. But that's just personal opinion no need to change it

Reply:
Fixed